src/GiraffeGenerator.Core/OpenApi.fs

```diff
@@ -75,20 +76,84 @@ type PrimTypeKind =
         | Some "boolean", _ -> Bool
         | Some "string", maybeStringFormat -> String(StringFormat.Parse maybeStringFormat)
         | _ -> failwithf "Unexpected type: %s and format: %A" typeKind format
+        
+    member s.GetDefaultLiteral (value: IOpenApiPrimitive) =
+        let fw t = sprintf "%s literal has been found for non-%s type" t t |> failwith
+        let inline oneWay kind (value: ^v) (toDefault: ^t -> DefaultableKind) =
+            match s with
+            | x when x = kind ->
+                let v = (^v:(member Value: ^t) value)
+                toDefault v
+            | _ -> fw (value.GetType().Name)
+            
+        match value with
+        | :? OpenApiInteger as int ->
+            match s with
+            | Int -> DefaultableKind.Integer int.Value
+            | Long -> DefaultableKind.Long (int64 int.Value)
+            | _ -> fw "integer"
+        | :? OpenApiLong as long ->
+            match s with
+            | Int ->  DefaultableKind.Long (int64 long.Value) 
+            | Long -> DefaultableKind.Long long.Value
+            | _ -> fw "int64"
+        | :? OpenApiDouble as double -> oneWay Double double DefaultableKind.Double
+        | :? OpenApiBoolean as bool -> oneWay Bool bool DefaultableKind.Boolean
+        | :? OpenApiDateTime as dateTime -> oneWay (String DateTimeString) dateTime DefaultableKind.DateTime
+        | :? OpenApiDate as date -> oneWay (String DateString) date DefaultableKind.Date
+        | :? OpenApiPassword as pwd -> oneWay (String PasswordString) pwd DefaultableKind.String
+        | :? OpenApiString as str ->
+            match s with
+            | String s ->
+                let inline oneWay x = oneWay (String s) x
+                match s with
+                | PasswordString
+                | StringFormat.String -> oneWay str DefaultableKind.String
+                | Custom "uri"
+                | Custom "uriref" -> oneWay {| Value = System.Uri str.Value |} DefaultableKind.Uri
+                | Custom "uuid"
+                | Custom "guid"
+                | Custom "uid" -> oneWay {| Value = Guid.Parse str.Value |} DefaultableKind.Guid
+                | _ -> oneWay str DefaultableKind.String
+            | _ -> fw "string"
+        | v -> failwith (sprintf "%A literals aren't supported for kind %A" v s)
+
+and DefaultableKind =
+    | String of string
+    | Integer of int
+    | Long of int64
+    | Date of DateTime
+    | DateTime of DateTimeOffset
```

Comment:
Do we need `DateTimeOffset` at all if all `dateX` types are being mapped to `DateTime`?

Reply:
Are they? As I can see in IntegrationTests/SpecWithParametersAndBodySchemas, OpenAPI `date` format is being mapped to `DateTime` and `date-time` format is being mapped to `DateTimeOffset`.

Reply:
Well, I could be wrong also :D
Are they? :)

Reply:
For now, yes. There going to be more variants with NodaTime support added =)